.eslintrc.js

```diff
@@ -280,6 +280,54 @@ module.exports = {
         },
       },
     },
+
+    /**
+     * TypeScript React-specific code
+     *
+     * Similar to above, but marks a majority of errors to warnings.
+     * TODO - combine rulesets and resolve actual errors
+     */
+    {
+      files: ['ui/**/*.ts', 'ui/**/*.tsx'],
+      extends: ['plugin:react/recommended', 'plugin:react-hooks/recommended'],
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+      plugins: ['react'],
+      rules: {
+        'react/no-unused-prop-types': 'warn',
+        'react/no-unused-state': 'warn',
+        'react/jsx-boolean-value': 'warn',
+        'react/jsx-curly-brace-presence': [
+          'warn',
+          {
+            props: 'never',
+            children: 'never',
+          },
+        ],
+        'react/no-deprecated': 'warn',
+        'react/default-props-match-prop-types': 'warn',
+        'react/jsx-no-duplicate-props': 'warn',
+        'react/display-name': 'off',
+        'react/no-unescaped-entities': 'warn',
+        'react/prop-types': 'off',
+        'react/no-children-prop': 'off',
+        'react/jsx-key': 'warn', // FIXME
+        'react-hooks/rules-of-hooks': 'warn', // FIXME
```

Comment:
27 instances to fix.
10 are specifically on calling hooks conditionally (which we should avoid)